src/Features/Core/Portable/UseCollectionInitializer/AbstractUseCollectionInitializerDiagnosticAnalyzer.cs

```diff
@@ -145,257 +156,64 @@ private void AnalyzeNode(SyntaxNodeAnalysisContext context, INamedTypeSymbol ien
-
-        internal ImmutableArray<TExpressionStatementSyntax> Analyze()
-        {
-            if (_syntaxFacts.GetObjectCreationInitializer(_objectCreationExpression) != null)
-            {
-                // Don't bother if this already has an initializer.
-                return ImmutableArray<TExpressionStatementSyntax>.Empty;
-            }
-
-            _containingStatement = _objectCreationExpression.FirstAncestorOrSelf<TStatementSyntax>();
-            if (_containingStatement == null)
-            {
-                return ImmutableArray<TExpressionStatementSyntax>.Empty;
-            }
-
-            if (!TryInitializeVariableDeclarationCase() &&
-                !TryInitializeAssignmentCase())
-            {
-                return ImmutableArray<TExpressionStatementSyntax>.Empty;
-            }
-
-            var matches = ArrayBuilder<TExpressionStatementSyntax>.GetInstance();
-            AddMatches(matches);
-            return matches.ToImmutableAndFree(); ;
-        }
-
-        private void AddMatches(ArrayBuilder<TExpressionStatementSyntax> matches)
-        {
-            var containingBlock = _containingStatement.Parent;
-            var foundStatement = false;
-
-            var seenInvocation = false;
-            var seenIndexAssignment = false;
-            
-            foreach (var child in containingBlock.ChildNodesAndTokens())
-            {
-                if (!foundStatement)
-                {
-                    if (child == _containingStatement)
-                    {
-                        foundStatement = true;
-                    }
-
-                    continue;
-                }
-
-                if (child.IsToken)
-                {
-                    return;
-                }
-
-                var statement = child.AsNode() as TExpressionStatementSyntax;
-                if (statement == null)
-                {
-                    return;
-                }
-
-                SyntaxNode instance = null;
-                if (!seenIndexAssignment)
-                {
-                    if (TryAnalyzeAddInvocation(statement, out instance))
-                    {
-                        seenInvocation = true;
-                    }
-                }
-
-                if (!seenInvocation)
-                {
-                    if (TryAnalyzeIndexAssignment(statement, out instance))
-                    {
-                        seenIndexAssignment = true;
-                    }
-                }
-
-                if (instance == null)
-                {
-                    return;
-                }
-
-                if (!ValuePatternMatches((TExpressionSyntax)instance))
-                {
-                    return;
-                }
-
-                matches.Add(statement);
-            }
-        }
 
-        private bool TryAnalyzeIndexAssignment(
-            TExpressionStatementSyntax statement,
-            out SyntaxNode instance)
+        public Task FixAllAsync(
```

Comment:
Why is this on the analyzer?  Shouldn't the  fixall implementation be on the code fix?